### ISSUE TITLE
Add latest OpenMPI versions

### ISSUE THREAD

**Issue by kuenishi** [chainer/chainer#5740]
[OpenMPI 4.0](https://www.open-mpi.org/software/ompi/v4.0/) is out so we have to test it out and decide whether we support or not. Also check 3.0 and 3.1 status.

c.f. https://github.com/chainer/chainermn/issues/221 Original issue with OpenMPI 3.0 https://github.com/open-mpi/ompi/issues/3972 has been solved so not it may work (or not?).

**Comment by belldandyxtq** [comment]
Testing the issus with different openmpi versions

- **Test Environment**
   CUDA 9.2
   GCC 5.4.0
   Infiniband FDR (56Gb/s) * 2

- **Compile Options**
./configure --prefix xxx --with-cuda --with-ucx path/to/ucx-1.5.0

- **Code Link**
    https://github.com/open-mpi/ompi/issues/3972

- **Commands**
    mpirun -np 2 gpu_test 1000

- **Openmpi Versions**
```
2.1.6				successful
2.1.6 + ucx 			successful
3.0.0				x (segmentation fault -> different from the report)
3.0.0 + ucx 			successful
3.1.0				x (frozen)
3.1.0 + ucx 			successful
3.1.3				x (frozen)
3.1.3 + ucx 			successful
4.0.0				successful (cannot compile without UCX)
```

**Comment by keisukefukuda** [comment]
Thanks. I will double-check the results. 
Further questions and issues:
 * Need to reproduce the whole results
 * What is the default PML layer of the versions of Open MPI?

**Comment by keisukefukuda** [comment]
I checked (1) MNIST and (2) Communicator test for Open MPI versions.

```
Open MPI version      MNIST      comm_test
openmpi-2.1.0         -          -
openmpi-2.1.1         -          -
openmpi-2.1.2         -          -
openmpi-2.1.3         -          -
openmpi-2.1.5         -          -
openmpi-2.1.6         -          -
openmpi-3.0.0         Bad        -
openmpi-3.0.1         Bad        -
openmpi-3.0.2         Bad        -
openmpi-3.0.3         -          -
openmpi-3.0.4         -          -
openmpi-3.1.0         Bad        -
openmpi-3.1.1         Bad        -
openmpi-3.1.2         Bad        -
openmpi-3.1.3         -          -
openmpi-3.1.4         -          -

```
`-`: OK

Basically, the problem was fixed in `3.0.3` and `3.1.3` for `3.0` and `3.1` branches respectively.

BTW, Open MPI 4.x is not supported by mpi4py. Thus, I will create a PR to update docs to explain the fact above.
